cloud/openstack/os_keystone_service.py

```diff
@@ -51,32 +50,43 @@
         - Indicate desired state of the resource
      choices: ['present', 'absent']
      default: present
-requirements: ["shade"]
+requirements:
+    - "python >= 2.6"
+    - "shade"
 author: Davide Guerri
 '''
 
 EXAMPLES = '''
 # Add Glance service
-- os_keystone_service: >
-    name=glance
-    service_type=image
-    description="Glance image service"
+- os_keystone_service:
+    name: glance
+    service_type: image
+    description: "Glance image service"
     cloud: dguerri
 
 # Delete Glance service
-- os_keystone_service: >
-    name=glance
-    service_type=image
-    description="Glance image service"
-    state=absent
+- os_keystone_service:
+    name: glance
+    service_type: image
+    description: "Glance image service"
+    state:absent
     cloud: dguerri
 '''
 
+def _system_state_change(module, service, new_service):
+    state = module.params['state']
+    if state == 'present' and \
+       (service is None or compare_service(service, new_service)):
+        return True
+    if state == 'absent' and service is not None:
+        return True
+    return False
+
 
 def compare_services(service_a, service_b):
-    return service_a['name'] == service_b['name'] and \
-           service_a['type'] == service_b['service_type'] and \
-           service_a['description'] == service_b['description']
+    return service_a.name == service_b['name'] and \
+           service_a.type == service_b['service_type'] and \
+           service_a.description == service_b['description']
```

Comment:
My bad, this is a change that slipped through when I was messing around with the code. Will fix!